src/Database/QueryBuilder/PgsqlQueryBuilder.php

```diff
@@ -54,11 +54,11 @@ class PgsqlQueryBuilder extends CommonQueryBuilder implements QueryBuilderInterf
         Column::TYPE_STRING => 'varchar',
     ];
 
-    private $adapter;
+    private $structure;
 
-    public function __construct(AdapterInterface $adapter = null)
+    public function __construct(Structure $structure)
```

Comment:
I was thinking about this and it would be better to use adapter. What if we will need some data from tables?